src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/Interop/MsQuicTlsSupportHelper.Windows.cs

```diff
@@ -0,0 +1,26 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using Microsoft.Win32;
+
+namespace System.Net.Quic.Implementations.MsQuic.Internal
+{
+    internal static partial class MsQuicTlsSupportHelper
+    {
+        private const string SChannelTLS1_3RegKey = @"SYSTEM\CurrentControlSet\Control\SecurityProviders\SCHANNEL\Protocols\TLS 1.3";
```

Comment:
BTW, in my quick search, this looks like the first registry check in product code we'd have, so @wfurt could you give it a look as well?

Is this what we want? Is there a way to get a reasonable error from OS when trying to use msquic and TSL 1.3 instead. How often this'll happen (meaning people disabling TLS 1.3 in registry)?

Reply:
One alternative I had thought of was a Windows only case for `MsQuicStatusCodes.GetError`, however, I thought that could be unwise as the error code `0x80090331` can occur in other cases that the client and server do not support matching algorithms.

Reply:
I think looking at the registry would be ok if we do one only at the beginning. One can still disable the protocols (or all it's algorithms) during application run. Schannel would react to it and we won't but I think that is ok. In cases like that, the user/admin are probably aware of it.